pkg/statsd/handler_cloud.go

```diff
@@ -192,12 +203,11 @@ func (ch *CloudHandler) Run(ctx context.Context) {
 }
 
 func (ch *CloudHandler) handleInstanceInfo(ctx context.Context, info gostatsd.InstanceInfo) {
-	metrics := ch.awaitingMetrics[info.IP]
-	if len(metrics) > 0 {
+	mm := ch.awaitingMetrics[info.IP]
```

Comment:
Someone plz double check my logic here:
- Metric comes in, goes to `handleIncomingMetrics`
- statsMetricHostsQueued is incremented because the host isn't in `awaitingMetrics` or `awaitingEvents`
- Event comes in, goes to `handleIncomingEvent`
- statsEventHostsQueued is **not** incremented because the host is in `awaitingMetrics`
- Lookup completes
- `handleInstanceInfo` sees the IP in `awaitingMetrics`, decrements `statsMetricHostsQueued`
- `handleInstanceInfo` sees the IP in `awaitingEvents`, decrements `statsEventHostsQueued `, it's now `-1`

I think the situation occurred in both the old logic, and this new code?  I'm inclined to think we should just drop both and use a single `hostsQueued` (or even just a `len(toLookupIPs)`)?

Reply:
This potentially problematic logic can be ignored, there's a refactor queued up which addresses it.

Reply:
I look forward to the update.

Reply:
I look forward to merging my third tweaking of this branch in to this once this branch is merged to master :okay: